### PR TITLE
issue #4660: ccUTF8 uses utf8-utf16 conversion in llvm3.4

### PR DESCRIPTION
**Pull request by dumganhar**

### CODE REVIEW

File: external/ConvertUTF/ConvertUTF.c [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+ * This file is distributed under the University of Illinois Open Source
+ * License. See LICENSE.TXT for details.
+ *
+ *===------------------------------------------------------------------------=*/
+/*
+ * Copyright 2001-2004 Unicode, Inc.
+ * 
+ * Disclaimer
+ * 
+ * This source code is provided as is by Unicode, Inc. No claims are
```
Comment (dumganhar) [inline external/ConvertUTF/ConvertUTF.c:14]: New license here.

**Comment by Seasons7** [comment]
@dumganhar 
Thank you for your implementation for UTF8/16 conversion.
I think your suggestion and implementation is good for all developers about license( avoid LGPL ) and stability.

I appreciate your review and point to code issue.
We are happy to use our test code for cocos2d-x.

**Comment by dumganhar** [comment]
@Seasons7
I modified a bit from your utf8/16 test case. Please refer to https://github.com/dumganhar/cocos2d-x/commit/945ad2eb04c11fdee3428832efcf40ed15784a9c

**Comment by Seasons7** [comment]
@dumganhar 
I've checked your modified code. Thank you very much for using our test code :)

BTW, Jenkins outputs Failed error.
http://115.28.134.83:8000/job/cocos-2dx-pull-request-build/3109/console

**Comment by dumganhar** [comment]
@Seasons7 , complication error was fixed. ci build succeed now. :)

**Comment by Seasons7** [comment]
@dumganhar , Thanks :)

**Pull request merged**